### PR TITLE
MDB: Make type warnings to respect Prisma typing

### PR DESCRIPTION
**Pull request by pimeys**
Mostly we just output Prisma types. Some typing, such as `Date/Timestamp`, ObjectId etc. will have both in the comments/warnings.

Closes: https://github.com/prisma/prisma/issues/12398

### CODE REVIEW

File: introspection-engine/connectors/mongodb-introspection-connector/tests/dirty_data/mod.rs [hunk 1]
```diff
@@ -103,7 +103,7 @@ fn mixing_types_with_the_same_base_type() {
     let expected = expect![[r#"
         model A {
           id    String @id @default(auto()) @map("_id") @db.ObjectId
-          /// Multiple data types found: Date: 50%, Timestamp: 50% out of 3 sampled entries
+          /// Multiple data types found: DateTime (Date): 50%, DateTime (Timestamp): 50% out of 3 sampled entries
```
Comment (janpio) [inline introspection-engine/connectors/mongodb-introspection-connector/tests/dirty_data/mod.rs:106]: Do we even have to use `Json` here if both would map to `DateTime` in Prisma?

Comment (pimeys) [reply introspection-engine/connectors/mongodb-introspection-connector/tests/dirty_data/mod.rs:106]: Kind of... The parsing logic is different for date and timestamp...

Comment (pimeys) [reply introspection-engine/connectors/mongodb-introspection-connector/tests/dirty_data/mod.rs:106]: Right now it's safer to use Json...

Comment (pimeys) [reply introspection-engine/connectors/mongodb-introspection-connector/tests/dirty_data/mod.rs:106]: [`ba94f7f` (#2787)](https://github.com/prisma/prisma-engines/pull/2787/commits/ba94f7f30349a0b0bf2b8f69dca3288b236186ed)

Comment (pimeys) [reply introspection-engine/connectors/mongodb-introspection-connector/tests/dirty_data/mod.rs:106]: nvm

**Review by tomhoule — APPROVED** [review]

**Pull request merged**